cmd/zed/lake/flags.go

```diff
@@ -31,31 +31,31 @@ func (f *Flags) SetFlags(set *flag.FlagSet) {
 	set.BoolVar(&f.Quiet, "q", false, "quiet mode")
 }
 
-func (f *Flags) RootPath() (iosrc.URI, error) {
-	return iosrc.ParseURI(f.Root)
+func (f *Flags) RootPath() (*storage.URI, error) {
+	return storage.ParseURI(f.Root)
 }
 
-func (f *Flags) Create(ctx context.Context) (*lake.Root, error) {
+func (f *Flags) Create(ctx context.Context, engine storage.Engine) (*lake.Root, error) {
```

Comment:
I would leave this logic to the commands instead of the shared Flags type.  For example, zed lake serve does not directly create a storage engine as service.Core allocates it as a remote engine.